p2p/libp2p/directSender_test.go

```diff
@@ -564,3 +649,34 @@ func TestDirectSender_ProcessReceivedDirectMessageFromMismatchesFromConnectedPee
 
 	assert.True(t, errors.Is(err, p2p.ErrInvalidValue))
 }
+
+func TestDirectSender_ProcessReceivedDirectMessageSignatureFails(t *testing.T) {
+	verifyCalled := false
+	expectedErr := fmt.Errorf("expected error")
+	ds, _ := libp2p.NewDirectSender(
+		context.Background(),
+		generateHostStub(),
+		blankMessageHandler,
+		&mock.P2PSignerStub{
```

Comment:
do we have a test with real signer?

Reply:
yes, in netMessenger as a semi-integration test
TestLibp2pMessenger_SendDirectWithRealMessengersShouldWork
TestLibp2pMessenger_SendDirectWithRealMessengersWithoutSignatureShouldWork